scripts/migrations/migrate_2.3.0.js

```diff
@@ -0,0 +1,104 @@
+const shell = require("shelljs");
+const { readContracts } = require("../util/utils.js");
+const hre = require("hardhat");
+const ethers = hre.ethers;
+const network = hre.network.name;
+// const { getStateModifyingFunctionsHashes } = require("../../scripts/util/diamond-utils.js");
+const tag = "HEAD";
+const version = "2.3.0";
+
+const config = {
+  // status at 451dc3d. ToDo: update this to the latest commit
+  addOrUpgrade: [
+    "DisputeResolverHandlerFacet",
+    "FundsHandlerFacet",
+    "MetaTransactionsHandlerFacet",
+    "OfferHandlerFacet",
+    "OrchestrationHandlerFacet1",
+    "ProtocolInitializationHandlerFacet",
+    "SellerHandlerFacet",
+    "TwinHandlerFacet",
+  ],
+  remove: [],
+  skipSelectors: {},
+  facetsToInit: {},
+  initializationData: "0x",
+};
+
+async function migrate(env) {
+  console.log(`Migration ${tag} started`);
+  try {
+    console.log("Removing any local changes before upgrading");
+    shell.exec(`git reset @{u}`);
+    const statusOutput = shell.exec("git status -s -uno scripts");
+
+    if (statusOutput.stdout) {
+      throw new Error("Local changes found. Please stash them before upgrading");
+    }
+
+    console.log("Installing dependencies");
+    shell.exec(`npm install`);
+
+    const { chainId } = await ethers.provider.getNetwork();
+    const contractsFile = readContracts(chainId, network, env);
+    if (contractsFile?.protocolVersion != "2.2.1") {
+      throw new Error("Current contract version must be 2.2.1");
```

Comment:
This check should be done at the very beginning of the migrate function (before any change has been done on the repo), so that if the criteria is not met, the repo is not modified

Reply:
I move to after the local changes checks otherwise, the version could be wrong if we only change the version locally.